src/script.rs

```diff
@@ -1,613 +1,45 @@
-    OP_RESERVED,
-    OP_VER,
-    OP_VERIF,
-    OP_VERNOTIF,
-    OP_RESERVED1,
-    OP_RESERVED2,
-    Unknown(u8),
-}
-
-impl From<u8> for Bad {
-    fn from(value: u8) -> Self {
-        match value {
-            0x50 => Bad::OP_RESERVED,
-            0x62 => Bad::OP_VER,
-            0x65 => Bad::OP_VERIF,
-            0x66 => Bad::OP_VERNOTIF,
-            0x89 => Bad::OP_RESERVED1,
-            0x8a => Bad::OP_RESERVED2,
-            _ => Bad::Unknown(value),
-        }
-    }
-}
-
-impl From<Bad> for u8 {
-    fn from(value: Bad) -> Self {
-        match value {
-            Bad::OP_RESERVED => 0x50,
-            Bad::OP_VER => 0x62,
-            Bad::OP_VERIF => 0x65,
-            Bad::OP_VERNOTIF => 0x66,
-            Bad::OP_RESERVED1 => 0x89,
-            Bad::OP_RESERVED2 => 0x8a,
-            Bad::Unknown(byte) => byte,
-        }
-    }
-}
-
-enum_from_primitive! {
-#[derive(Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Debug)]
-#[repr(u8)]
-pub enum Disabled {
-    // splice ops
-    OP_CAT = 0x7e,
-    OP_SUBSTR = 0x7f,
-    OP_LEFT = 0x80,
-    OP_RIGHT = 0x81,
-    // bit logic
-    OP_INVERT = 0x83,
-    OP_AND = 0x84,
-    OP_OR = 0x85,
-    OP_XOR = 0x86,
-    // numeric
-    OP_2MUL = 0x8d,
-    OP_2DIV = 0x8e,
-    OP_MUL = 0x95,
-    OP_DIV = 0x96,
-    OP_MOD = 0x97,
-    OP_LSHIFT = 0x98,
-    OP_RSHIFT = 0x99,
-
-    //crypto
-    OP_CODESEPARATOR = 0xab,
-}
-}
-
-impl From<Disabled> for u8 {
-    fn from(value: Disabled) -> Self {
-        // This is how you get the discriminant, but using `as` everywhere is too much code smell
-        value as u8
-    }
-}
-
-/// When writing scripts, we don’t want to allow bad opcodes, so `Opcode` doesn’t include them.
-/// However, when validating scripts, bad opcodes only cause a failure when they’re on an active
-/// branch, so this type allows us to hold onto the bad opcodes when parsing.
-pub enum PossiblyBad {
-    Good(Opcode),
-    Bad(Bad),
-}
-
-pub struct ParsedOpcode<'a> {
-    /// The [`PossiblyBad`] allows us to preserve unknown opcodes, which only trigger a failure if
-    /// they’re on an active branch during interpretation.
-    pub opcode: PossiblyBad,
-    pub remaining_code: &'a [u8],
-}
+use crate::{
+    opcode::{
+        self,
+        push_value::{
+            LargeValue::*,
+            SmallValue::{self, *},
+        },
+        Operation::*,
+        PushValue,
+    },
+    script_error::ScriptError,
+    Opcode,
+};
 
 /// “Strict” scripts disallow [`Bad`] opcodes, and so we collect any bad opcodes that we find, so we
 /// can report all of them, rather than just the first one.
 pub enum StrictError {
     /// A script validation failure.
     Script(ScriptError),
     /// All of the [`Bad`] opcodes we found.
-    BadOpcodes(Vec<Bad>),
+    BadOpcodes(Vec<opcode::Bad>),
 }
 
 /** Serialized script, used inside transaction inputs and outputs */
 #[derive(Clone, Debug)]
-pub struct Script<'a>(pub &'a [u8]);
+pub struct Code<'a>(pub &'a [u8]);
```

Comment:
This is all gonna need changelog entries. But this second commit's refactor is significant enough that I'll accept an issue being opened for doing this in a follow-up PR after these last few refactor PRs have landed (as long as that issue is completed before the release this all goes out in, and we don't miss anything as a result).